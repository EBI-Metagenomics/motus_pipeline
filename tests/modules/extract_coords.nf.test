nextflow_process {

    name "Test module extract models from fasta"
    script "modules/extract_coords.nf"
    process "EXTRACT_MODELS"

    test("extract models") {

        when {
            params {
                name = "test_result"
                seqs = "${baseDir}/tests/modules/fixtures/extract_coords/test.fasta"
            }
            process {
                """
                input[0] = channel.value("${params.name}")
                input[1] = channel.fromPath("${params.seqs}", checkIfExists: true)
                """
            }
        }

        then {
            assert process.success
            // assert snapshot(process.out).match()
            with(process.out.ssu_fasta) {
                assert size() == 1
                assert get(0) ==~ ".*/test_result_SSU.fasta"
            }
            with(process.out.lsu_fasta) {
                assert size() == 1
                assert get(0) ==~ ".*/test_result_LSU.fasta"
            }
            with(process.out.seq_cat_folder) {
                assert get(0) ==~ ".*/sequence-categorisation"
            }
        }

    }
}