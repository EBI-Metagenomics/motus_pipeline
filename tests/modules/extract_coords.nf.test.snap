{
    "extract models": {
        "content": [
            {
                "ssu_fasta": [
                    "test_result_SSU.fasta:md5,d82551ffe1cbdcf745278eb09199de56"
                ],
                "lsu_fasta": [
                    "test_result_LSU.fasta:md5,a3687de0c0d784f4930bd9c780638bb3"
                ],
                "seq_cat_folder": [
                    [
                        [
                            "test_result_5_8S_rRNA_bacteria.fasta:md5,7dfc2985d28cca6db4c0ee2a26a92175"
                        ],
                        "test_result_LSU.fasta:md5,a3687de0c0d784f4930bd9c780638bb3",
                        "test_result_LSU_rRNA_bacteria.RF02541.fa:md5,a3687de0c0d784f4930bd9c780638bb3",
                        "test_result_SSU.fasta:md5,d82551ffe1cbdcf745278eb09199de56",
                        "test_result_SSU_rRNA_bacteria.RF00177.fa:md5,d82551ffe1cbdcf745278eb09199de56"
                    ]
                ]
            }
        ],
        "timestamp": "2023-03-24T13:22:00+0000"
    }
}